ft-sdk/src/session.rs

```diff
@@ -49,13 +49,64 @@ impl SessionID {
         self.update_one(conn, Some(user_id), None)
     }
 
-    /// Set the data for the given session.
-    pub fn set_data(
+    /// Get the session data object.
+    /// Useful for fetching the entire session data in a single db call. Use
+    /// [get_key](SessionID::get_key) instead if you only need a single key
+    pub fn data(
         &self,
         conn: &mut ft_sdk::Connection,
-        data: serde_json::Value,
-    ) -> Result<SessionID, ft_sdk::Error> {
-        self.update_one(conn, None, Some(data))
+    ) -> Result<serde_json::Value, diesel::result::Error> {
```

Comment:
Yet another option is we return `SessionData` from this function, which opaqueyly stores entire data, and provde `.get_key()` etc on it (and leave `.get_key()` here with current semantics as well, so we can can more than one key in one setting. Similarly we can hadd `.set_key()` and `SessionData.persist()`. We can even make it “update centric instead of full overwrite”, so eg on every `set_key()` we store the key and value that were updated, and on `.persist()` we get latest version of data from session table and only modify those keys that were updated so we do not lose data by another transaction.

Reply:
have tried an implementation in https://github.com/FifthTry/ft-sdk/pull/52/commits/05e055c6535b5ab33ce1877bc59d607fc9fb6fc9.

I have remove the `SessionID::get_key` as it was fetching all the data anyway, I could have modified the sql query to only fetch the requested key but I don't need this method nor could I see the use case if `SessionData::get_key` already exists.